src/Elastic.Apm/Helpers/CookieHeaderRedacter.cs

```diff
@@ -0,0 +1,206 @@
+// Licensed to Elasticsearch B.V under one or more agreements.
+// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information
+
+#if !NETFRAMEWORK
+using System;
+using System.Buffers;
+#endif
+using System.Collections.Generic;
+using System.Configuration;
+using System.Text;
+
+namespace Elastic.Apm.Helpers;
+
+internal static class CookieHeaderRedacter
+{
+#if !NETFRAMEWORK
+	private static ReadOnlySpan<char> Redacted => Consts.Redacted.AsSpan();
+	private static ReadOnlySpan<char> SemiColon => ";".AsSpan();
+	private static ReadOnlySpan<char> Comma => ",".AsSpan();
+#endif
+
+	public static string Redact(string cookieHeaderValue, IReadOnlyList<WildcardMatcher> namesToSanitize)
+	{
+		// Implementation notes:
+		// This method handles a cookie header value for both ASP.NET (classic) and
+		// ASP.NET Core. As a result it must handle two possible formats. In ASP.NET
+		// (classic) the string is the actual Cookie value as sent over the wire, conforming
+		// to the HTTP standards. This uses the semicolon separator and a space between
+		// entries. For ASP.NET Core, when we parse the headers, we convert from the
+		// StringValues by calling ToString. This results in each entry being separated
+		// by a regular colon and no space.
+
+		// When the headers are stored into the `Request.Headers` on our APM data model,
+		// multiple headers of the same name, are concatenated into a comma-separated list.
+		// When redacting, we preserve this separation.
+
+		if (string.IsNullOrEmpty(cookieHeaderValue))
+			return null;
+
+		var redactedCookieHeader = string.Empty;
+
+#if NETFRAMEWORK
+		// The use of `Span<T>` in NETFX can cause runtime assembly loading issues due to our friend,
+		// binding redirects. For now, we take a slightly less allocation efficient route here, rather
+		// than risk introducing runtime issues for consumers. Technically, this should be "fixed" in
+		// NET472+, but during testing I surprisingly still reproduced an exception. Elastic.APM depends on
+		// `System.Diagnostics.DiagnosticSource 5.0.0` which itself depends on `System.Runtime.CompilerServices.Unsafe`
+		// which is where the exception occurs. 5.0.0 is marked as deprecated so we could look to prefer
```

Comment:
Can you create an issue for this? Would love to bump SDD to 6.0.0 if we can :)

Reply:
https://github.com/elastic/apm-agent-dotnet/issues/2445